myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/JHSTaskFactoryImpl.java

```diff
@@ -0,0 +1,366 @@
+          String executorPathString = myriadExecutorConfiguration.getPath();
+          String sudoExecutorCmd = "sudo -E -u " + cfg.getFrameworkUser().get() + " -H " +
+              executorCmd;
+
+          //Concatenate all the subcommands
+          String cmd = tarCmd + "&&" + chownCmd + "&&" + configCopyCmd + "&&" + sudoExecutorCmd;
+
+          //get the nodemanagerURI
+          //We're going to extract ourselves, so setExtract is false
+          LOGGER.info("Getting Hadoop distribution from:" + nmURIString);
+          URI nmUri = URI.newBuilder().setValue(nmURIString).setExtract(false)
+              .build();
+
+          //get configs directly from resource manager
+          String configUrlString = getConfigurationUrl();
+          LOGGER.info("Getting config from:" + configUrlString);
+          URI configUri = URI.newBuilder().setValue(configUrlString)
+              .build();
+
+          //get the executor URI
+          LOGGER.info("Getting executor from:" + executorPathString);
+          URI executorUri = URI.newBuilder().setValue(executorPathString).setExecutable(true)
+              .build();
+
+          LOGGER.info("Slave will execute command:" + cmd);
+          commandInfo.addUris(nmUri).addUris(configUri).addUris(executorUri).setValue("echo \"" + cmd + "\";" + cmd);
+
+          commandInfo.setUser(cfg.getFrameworkSuperUser().get());
+
+        } else {
+          commandInfo.setValue(executorCmd);
+        }
+        return commandInfo.build();
+      }
+      
+      private static String getFileName(String uri) {
+        int lastSlash = uri.lastIndexOf('/');
+        if (lastSlash == -1) {
+          return uri;
+        } else {
+          String fileName = uri.substring(lastSlash + 1);
+          Preconditions.checkArgument(!Strings.isNullOrEmpty(fileName),
+              "URI should not have a slash at the end");
+          return fileName;
+        }
+      }
+
+      private String getConfigurationUrl() {
+        YarnConfiguration conf = new YarnConfiguration();
+        String httpPolicy = conf.get(YARN_HTTP_POLICY);
+        if (httpPolicy != null && httpPolicy.equals(YARN_HTTP_POLICY_HTTPS_ONLY)) {
+          String address = conf.get(YARN_RESOURCEMANAGER_WEBAPP_HTTPS_ADDRESS);
+          if (address == null || address.isEmpty()) {
+            address = conf.get(YARN_RESOURCEMANAGER_HOSTNAME) + ":8090";
+          }
+          return "https://" + address + "/conf";
+        } else {
+          String address = conf.get(YARN_RESOURCEMANAGER_WEBAPP_ADDRESS);
+          if (address == null || address.isEmpty()) {
+            address = conf.get(YARN_RESOURCEMANAGER_HOSTNAME) + ":8088";
+          }
+          return "http://" + address + "/conf";
+        }
+      }
+
+      /**
+       * Helper method to reserve ports
+       * @param offer
+       * @param requestedPorts
+       * @return
+       */
+      private List<Long> getAvailablePorts(Offer offer, List<Long> requestedPorts) {
```

Comment:
I am still thinking about this one. I may remove handling ports here altogether for now, as even if JHS knows about those ports, clients will not automatically know about it. It should be some mechanism of passing this knowledge to clients. Today it is mapred-site.xml.